### ISSUE TITLE
Improve AI classification: expand coverage, resolve conflicts, refine centroids

### ISSUE THREAD

**Issue by shotleybuilder** [fractalaw/fractalaw#14]
## Context

The centroid-based AI classification pipeline landed in `5c451b9` (#13). It classifies laws by domain, family, and subjects using mean-pooled section embeddings compared against label centroids. First-run results:

- **452 / 19,318 laws** classified (only those with embeddings in LanceDB)
- **74.6% agreement** with ground truth (302 confirmed, 103 conflicts, 47 predicted-only)
- **Mean family confidence: 0.859**
- **24 family centroids**, 3 domain centroids, 141 subject centroids

The pipeline works end-to-end but there are clear opportunities to improve coverage and accuracy.

### 1. Expand text/embedding coverage

Only 452 of 19,318 laws currently have text in LanceDB. Classification reach is directly bottlenecked by this. Ingesting text for more laws (especially the ~13K that already have ground-truth family labels) would:

- Increase the number of classifiable laws from 452 to potentially thousands
- Improve centroid quality by training on more examples per family
- Enable meaningful held-out evaluation (currently too few samples for robust cross-validation)

### 2. Resolve the 103 conflict cases

103 laws have `classification_status = 'conflict'` — the AI disagrees with the ground-truth family label. These are queryable:

```sql
SELECT name, family, classified_family, classification_confidence
FROM legislation
WHERE classification_status = 'conflict'
ORDER BY classification_confidence DESC
```

Common conflict patterns observed:

| AI predicts | Ground truth says | Notes |
|-------------|-------------------|-------|
| Climate Change | Energy | Overlapping regulatory domains |
| Fire / Dangerous Substances | Consumer Product Safety | Safety-adjacent families |
| Planning | Historic Environment | Planning acts with heritage provisions |

Some conflicts may indicate ground-truth mislabelling. A manual review pass would both correct labels and improve centroids.

### 3. Refine centroid quality

The 74.6% agreement rate is a solid baseline but leaves room for improvement:

- **Weighted centroids**: Weight law contributions by text length or recency rather than uniform averaging
- **k-NN fallback**: For low-confidence predictions (below p10 = 0.763), use k-nearest-neighbour voting instead of nearest-centroid
- **Hierarchical refinement**: Use domain classification to narrow the family candidate set before scoring
- **Exclude outliers**: Remove laws that are extreme outliers from their family centroid before recomputing

### 4. Subject revival for post-2013 laws

The `subjects` classification scheme was maintained 1987–2012 and abandoned mid-2013. The pipeline predicted subjects for 121 of 6,358 post-2013 laws — modest coverage limited by the 452-law embedding pool. With expanded text coverage (item 1), subject revival could reach thousands of post-2013 laws.

### 5. Sub-family classification

`sub_family` has only 7.5% ground-truth coverage (1,449 / 19,318 laws) — too sparse for reliable centroid training. Options:

- Derive sub_family from family + subjects heuristically
- Wait for more ground-truth labels from conflict resolution (item 2)
- Use a different approach (e.g., keyword matching within family groups)

## Non-goals (for now)

- Neural/fine-tuned classifiers (the centroid approach should be pushed further first)
- Per-section classification (labels are per-law by design)
- CSV re-parsing (labels come from the legislation table, not LAT/AMD CSVs directly)
